### PR TITLE
Process API addressable ports

### PR DESCRIPTION
**Pull request by bergie**
Improves test coverage for dealing with addressable ports with Process API. Needed for #526

### CODE REVIEW

**Comment by bergie** [comment]
@trustmaster would be good to give this an initial review round

File: src/lib/Component.coffee [hunk 1]
```diff
@@ -222,14 +225,40 @@ class Component extends EventEmitter
...
+  addToResult: (result, port, ip, before = false) ->
+    if port.indexOf('[') is -1
+      # Regular port
+      outport = port
+    else
+      [port, outport, idx] = port.match /(.*)\[([0-9+])\]/
+    method = if before then 'unshift' else 'push'
+    if @outPorts[outport].isAddressable()
+      idx = if idx then parseInt(idx) else ip.index
+      result[outport] = {} unless result[outport]
```
Comment (trustmaster) [inline src/lib/Component.coffee]: Why is `result[outport]` an object instead of array here?

File: src/lib/InPort.coffee [hunk 2]
```diff
@@ -32,6 +32,7 @@ class InPort extends BasePort
       @process = process
 
     if options.handle
+      platform.deprecated 'InPort handle callback is deprecated. Please use Process API'
```
Comment (trustmaster) [inline src/lib/InPort.coffee]: I thought we would keep that as low-level. But it's your call.

Comment (bergie) [inline src/lib/InPort.coffee]: I consider it somewhat dangerous, given that it doesn't actually read from the port buffer, and doesn't provide any activation/deactivation logic.

If users must, they can of course still do `component.inPorts.foo.on 'ip', (ip) -> # do custom stuff`

File: src/lib/Component.coffee (hunk 1 above)
Comment (bergie) [inline src/lib/Component.coffee]: @trustmaster because we populate array per port index under it

**Comment by bergie** [comment]
Follow-up would be to add convenience methods for checking / fetching input data by connection idx

**Pull request merged**